alert.go

```diff
@@ -563,6 +588,18 @@ func (a *AlertNode) determineLevel(now time.Time, fields models.Fields, tags map
 			continue
 		} else if pass {
 			return AlertLevel(l)
+		} else {
+			lrse := a.levelResets[l]
```

Comment:
God, that logic is terrible. Your thinking is much better. I need to write the tests now.